### PR TITLE
Replace 'lookbehind' regex pattern for Ireland 'country.name.en.regex'

### PR DESCRIPTION
**Pull request by etpinard**
Hi, 

First at all, thanks very much for this amazing package! Your country name regexes are simply amazing. 
At [plotly](https://plot.ly/), we use them to power our geographic maps visualisations e.g. this [chorolpleth](https://plot.ly/~etpinard/4666.embed).

Being web-based, the regexes are compiled in JavaScript which very unfortunately does not support all the same regex special characters as R. In particular, JavaScript does not support the _lookbehind_ pattern `?<!` as defined in current `master` data for Ireland. 

This PR substitute the _lookbehind_ for something a little more _ad hoc_ where `northern ireland` is explicitly skipped. This would allow us to continue using your country name regex patterns without patching the Ireland item.

### CODE REVIEW

**Comment by vincentarelbundock** [comment]
Happy to meet another montrealer on github. I really like the work that your team is doing at plotly too!

When I run the test suite with your new regex, I get this:

```
Failed -------------------------------------------------------------------------
1. Failure: a variety of country names yield correct iso3c codes (@test-regex-external.R#112) 
iso3c_of("Republic of Ireland") not equal to "IRL".
1/1 mismatches
x[1]: NA
y[1]: "IRL"
```

File: data/countrycode_data.csv [hunk 1]
```diff
@@ -114,7 +114,7 @@ ASIA,Asia,IND,750,,Asia,Asia/Pacific,100,IN,V[AEIO],V,534,IND,IN,IND,356,Souther
 ASIA,Asia,INS,850,,Asia,Asia/Pacific,101,ID,W[AIQR],W,536,INA,ID,IDN,360,South-Eastern Asia,360,IDN,إندونيسيا,Indonesien,indonesien,Indonesia,indonesia,Indonesia,Indonésie,Индонезия,印度尼西亚,ID,ID,IDN,INS,850.0,360
 MAF,Asia,IRN,630,,Middle East,Middle-East,102,IR,OI,O,429,IRI,IR,IRN,364,Southern Asia,364,IRN,إيران (جمهورية - إسلامية),Iran,\biran|persia,Iran (Islamic Republic of),\biran|persia,Irán (República Islámica del),Iran (République islamique d'),Иран (Исламская Республика),伊朗伊斯兰共和国,IR,IR,IRN,IRN,630.0,364
 MAF,Asia,IRQ,645,,Middle East,Middle-East,103,IZ,OR,O,433,IRQ,IQ,IRQ,368,Western Asia,368,IRQ,العراق,Irak,\birak|mesopotamia,Iraq,\biraq|mesopotamia,Iraq,Iraq,Ирак,伊拉克,IQ,IQ,IRQ,IRQ,645.0,368
-OECD1990,Europe,IRE,205,EU,Eurocontrol,ESRA North-West,104,EI,EI,E,178,IRL,IE,IRL,372,Northern Europe,372,IRL,أيرلندا,Irland,irland,Ireland,^(?!northern\sireland)ireland,Irlanda,Irlande,Ирландия,爱尔兰,IE,IE,IRL,IRE,205.0,372
+OECD1990,Europe,IRE,205,EU,Eurocontrol,ESRA North-West,104,EI,EI,E,178,IRL,IE,IRL,372,Northern Europe,372,IRL,أيرلندا,Irland,irland,Ireland,(^ireland)|(^republic.*ireland),Irlanda,Irlande,Ирландия,爱尔兰,IE,IE,IRL,IRE,205.0,372
```
Comment (etpinard) [inline data/countrycode_data.csv]: Far from the cleanest solution, but this gets all the tests passing.

**Comment by vincentarelbundock** [comment]
Looks great, and thanks for the test too.

Just to clarify, @etpinard , do you need a CRAN release for this, or are you happy with running the github devel version? I usually prefer releasing bundles of changes to CRAN, since that process can sometimes be a pain.

**Comment by etpinard** [comment]
Thanks! No need for a CRAN release. Getting this on the GitHub master will suffice.

**Comment by cjyetman** [comment]
Because the internal function of `countrycode` does not strip leading and trailing whitespace (by design), this causes a a problem...
```r
> countrycode::countrycode('  Ireland', 'country.name', 'country.name')
[1] NA
Warning message:
In countrycode::countrycode("  Ireland", "country.name", "country.name") :
  Some values were not matched unambiguously:   Ireland
```
@etpinard is there some way to adjust this regex so that it works for you but also deals with leading and trailing whitespace? maybe `(^\sireland)|(^\srepublic.*ireland)`?

**Pull request merged**